Graphics/GraphicsEngineOpenGL/src/RenderDeviceGLImpl.cpp

```diff
@@ -1687,11 +1687,23 @@ void RenderDeviceGLImpl::PurgeContextCaches(GLContext::NativeGLContextType Conte
 {
     {
         Threading::SpinLockGuard FBOCacheGuard{m_FBOCacheLock};
-        m_FBOCache.erase(Context);
+
+        auto it = m_FBOCache.find(Context);
+        if (it != m_FBOCache.end())
+        {
+            it->second.Clear();
+            m_FBOCache.erase(it);
+        }
     }
     {
         Threading::SpinLockGuard VAOCacheGuard{m_VAOCacheLock};
-        m_VAOCache.erase(Context);
+
+        auto it = m_VAOCache.find(Context);
+        if (it != m_VAOCache.end())
+        {
+            it->second.Clear();
```

Comment:
Same here - when VAO cache is destroyed in the next line, all objects it holds should be released.

Reply:
If users `PurgeCaches`, force clearing  to avoid assertion errors in destructors of VAOCache and FBOCache.

So if VAO cache is destroyed automatically, and some internal VAO/FBO state cache bugs occurs, the assertion in its destructor can still check for possible errors.